azext_iot/operations/hub.py

```diff
@@ -2188,14 +2188,9 @@ def http_wrap(target, device_id, generator):
             )
             client_mqtt.execute(data=generator(), properties=properties_to_send, publish_delay=msg_interval, msg_count=msg_count)
         else:
-            six.print_("Sending and receiving events via https")
-            token, op = execute_onthread(
-                method=http_wrap,
-                args=[target, device_id, generator()],
-                interval=msg_interval,
-                max_runs=msg_count,
-                return_handle=True,
-            )
+            op = Thread(target=http_wrap, args=(target, device_id, generator(), msg_interval, msg_count))
```

Comment:
Could we not have improved `execute_onthread` rather than use `Thread` and `Event` directly here? BTW `execute_onthread` is used a lot in tests

Reply:
The other option was to introduce tqdm in the execute_onthread function, which is something I didn't want to do because TQDM may not be useful every time execute_onthread is called